src/components/input-with-icon/InputWithIcon.jsx

```diff
@@ -7,9 +7,9 @@ import { styles } from '~/components/input-with-icon/InputWithIcon.styles'
 
 const InputWithIcon = ({ value, onClear, startIcon, sx, ...props }) => {
   return (
-    <Box sx={[styles.root, sx]}>
+    <Box data-testid='inputWithIconContainer' sx={[styles.root, sx]}>
       {startIcon}
-      <InputBase sx={styles.input} value={value} {...props} />
+      <InputBase data-testid='inputWithIconInputBase' sx={styles.input} value={value} {...props} />
```

Comment:
Sonar do not like this row,fix it, please

Reply:
Thank you for reviewing the tests report and for bringing to my attention that Sonar does not accept the attribute data-testid='inputWithIconInputBase'. I discovered that the InputBase component from MUI has the role 'textbox':

Here are the accessible roles:
  textbox:
  Name "":
  <input
    class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
    type="text"
    value="value"
  />

As a result, I have updated the testing approach from using getByTestId to getByRole to avoid the use of the data-testid='inputWithIconInputBase' attribute.